lib/src/native/realm_core.dart

```diff
@@ -326,6 +326,47 @@ class _RealmCore {
     return RealmResultsHandle._(pointer);
   }
 
+  RealmResultsHandle queryTable(Realm realm, int classKey, String query, List<Object> args) {
+    return using((arena) {
+      final length = args.length;
+      final argsPointer = arena<Pointer<realm_value_t>>(length);
+      for (var i = 0; i < length; ++i) {
+        argsPointer[0] = _toRealmValue(args[i], arena);
```

Comment:
```suggestion
        argsPointer[i] = _toRealmValue(args[i], arena);
```

Reply:
This won't do when we change type of argsPointer. Instead I introduce a `_intoRealmValue` method contruct the realm value in place.

Reply:
yeah it wont fix it. what was I thinking.